hydra-node/exe/tx-cost/TxCost.hs

```diff
@@ -190,8 +190,8 @@ computeAbortCost =
 
 computeFanOutCost :: IO [(NumUTxO, TxSize, MemUnit, CpuUnit, Lovelace)]
 computeFanOutCost = do
-  interesting <- catMaybes <$> mapM compute [1, 2, 3, 5, 10, 20]
-  limit <- maybeToList . getFirst <$> foldMapM (fmap First . compute) [100, 99 .. 21]
+  interesting <- catMaybes <$> mapM compute [1, 2, 3, 5, 10, 50]
+  limit <- maybeToList . getFirst <$> foldMapM (fmap First . compute) [100, 99 .. 51]
```

Comment:
@pgrange Why did you revert this? This DID have an effect as it was running the transaction cost benchmarks with <50 assets to see what the upper bound right now is (roughly 49).

Reply:
That was my next item in my todo-list: ask you what's the purpose of this change.

As of yesterday, I've noticed that the test failed for size problems and then you pushed this change so I guessed that it was related to that and I wanted to check that we could pass the test without it.

Now I'm happy to understand what's the purpose of the change?

Reply:
It will update the benchmark on this PR to get a ballpark estimate of what the maximum number of assets supported to fan out is right now. For example:


| UTxO  | Tx size | % max Mem | % max CPU | Min fee ₳ |
| :---- | ------: | --------: | --------: | --------: |
...
| 49| 16374 | 83.44 | 46.74 | 1.88 |

So this can be used to have the failing test (in StateSpec) not try to fit too many assets into one 16kB transaction - there is a hard-coded upper bound used to generate tests https://github.com/input-output-hk/hydra/blob/08e155bdaf871a72bf3aedf6ce73796a5f6b9298/hydra-node/test/Hydra/Chain/Direct/StateSpec.hs#L392.

Reply:
> So this can be used to have the failing test (in StateSpec)

OK then maybe that's the point. I did not observer any failing test in the last push. But maybe I'm missing something as this bench part is not clear to me yet.